x-pack/filebeat/input/http_endpoint/crc.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package http_endpoint
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Validator func(crcValidator, mapstr.M) (int, string, error)
+
+type crcValidator struct {
+	provider  string    // Name of the webhook provider
+	key       string    // Key to identify CRC requests (optional)
+	value     string    // Value of the field to identify CRC requests (optional)
+	challenge string    // Key of the challenge token
+	secret    string    // Webhook's secret token
+	validator Validator // Function to process the CRC request
+	output    mapstr.M  // Output JSON template
+}
+
+// Create new CRC handler based in the webhook provider
+func newCRC(crcProvider string, secret string) crcValidator {
+	var newCRC crcValidator
+	switch strings.ToLower(crcProvider) {
+	case "zoom":
+		newCRC = newZoomCRC(secret)
+	default:
+		// Do nothing
+	}
+	return newCRC
+}
+
+// Initialize CRC struct for Zoom provider
```

Comment:
Conventional docs for this would be something like
```suggestion
// newZoomCRC returns a CRC handler for the Zoom API.
```